network/peer.go

```diff
@@ -836,3 +879,45 @@ func (p *peer) getIP() utils.IPDesc {
 	defer p.ipLock.RUnlock()
 	return p.ip
 }
+
+// addAlias marks that we have found another
+// IP that we can connect to this peer at.
+//
+// assumes stateLock is held
+func (p *peer) addAlias(ip utils.IPDesc) {
+	p.ipLock.Lock()
+	defer p.ipLock.Unlock()
+
+	p.net.peerAliasIPs[ip.String()] = struct{}{}
+	p.aliases = append(p.aliases, alias{
+		ip:    ip,
+		added: p.net.clock.Time(),
+	})
+}
+
+// releaseAliases removes enables network to
+// reconnect on aliases that have timed out. To
+// force all aliases to be removed, the caller
+// should set now to be the zero time.
+func (p *peer) releaseAliases(now time.Time, holdsStateLock bool) {
+	p.ipLock.Lock()
```

Comment:
This doesn't seem to be needed. You are not accessing peer.ip..  And f you do you should use getIP()

Reply:
I utilize ipLock as a guard for aliases as well. Open to adding another lock if you prefer. Note the comment here: https://github.com/ava-labs/avalanchego/pull/746/files#diff-b06a14805cb9806d82ef18d987b32228ab9b36a4e679c7c128efd711038991b2R69

Reply:
Would probably make more sense to do so.

Reply:
Yea,  didn't notice it was for aliases... A different lock could reduce contention.  I would consider a lock for aliases.

IpLock is s RWMutex... 
It could be a bit more optimal to use the RLock to check if p.aliases has a length as it would be non blocking..
And use a write lock only if the p.aliases has a length..

Reply:
In this case, we'd want to hold the `aliasesLock` during check and the removal (if necessary) otherwise we may attempt to access an item at 0 that is no longer there (because the lock was acquired by a different goroutine that removed an item). Nonetheless, separating the locks now!

Reply:
You could do a double check inside the write lock to confirm the length.

if len > 0 {
   p.ipLock.Lock()
   defer p.ipLock.Unlock()
   if len(p.aliases) == 0 {
      return
   }
   
  ......

Reply:
I'm not sure but my gut says the overhead of acquiring and releasing the alias lock to check length and then acquiring it again to check length and modify `p.aliases` is worse than just holding it until after we've made the modification:
```
p.aliasLock.Lock()
if len(p.aliases) == 0 {
  p.aliasLock.Unlock()
  continue
}

next := p.aliases[0]
p.aliases = p.aliases[1:]
p.aliasLock.Unlock()
```

Reply:
I actually think the way I was doing it before with `defer` could've caused a deadlock on `addAlias`. Thanks again for catching.